server/src/main/java/io/pravega/schemaregistry/server/rest/auth/AuthHandlerManager.java

```diff
@@ -135,25 +162,25 @@ public boolean authorize(String resource, AuthHandler.Permissions level) {
             Preconditions.checkNotNull(principal.get(), "Authentication should have been called before authorization");
             return handler.authorize(resource, principal.get()).ordinal() >= level.ordinal();
         }
+        
+        @Override
+        public Principal getUserPrincipal() {
+            return principal.get();
+        }
 
-        /**
-         * API to authenticate and authorize access to a given resource.
-         *
-         * @param resource    The resource identifier for which the access needs to be controlled.
-         * @param level       Expected level of access.
-         * @return Returns true if the entity represented by the custom auth headers had given level of access to the resource.
-         * Returns false if the entity does not have access.
-         * @throws AuthenticationException if an authentication failure occurred.
-         */
-        public boolean authenticateAndAuthorize(String resource, AuthHandler.Permissions level) throws AuthenticationException {
-            boolean retVal = false;
-            try {
-                authenticate();
-                authorize(resource, level);
-            } catch (AuthException e) {
-                throw new AuthenticationException("Authentication failure");
-            }
-            return retVal;
+        @Override
+        public boolean isUserInRole(String role) {
+            return false;
+        }
+
+        @Override
+        public boolean isSecure() {
```

Comment:
This method is supposed to return true if the request was made through a secure channel/HTTPS:  https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/SecurityContext.html#isSecure--

So, throwing false always is incorrect. If the implementation of this method is not needed, we should instead just throw `UnsupportedOperationException`.